### ISSUE TITLE
Consider backbone a failed experiment

### ISSUE THREAD

**Issue by hampelm** [detroitledger/detroitledger#61]
For discussion purposes: 

Pros of a backbone-based front end: 
- Speed and simplicity of development (don't need to change app code to get additional data, for example)  
- Robust and simple to serve (super easy to deploy static files) 

Cons: 
- Gross hacks needed for SEO (`escaped_fragment`, #56)
- Significant deploy complexity added (#49, bugs like #54, memory problems with #46)
- No longer just deploying static files
- SEO-friendly URLs are harder; can't just "turn on" pretty URLs (#53)
- Cascade of related problems

Basically, every reason why I pushed for backbone has been negated by Google limitations, and since we're a content-driven site, that's paramount. 

Maybe there's a front-end framework that solves some of these issues?

**Comment by hampelm** [comment]
I'm sure I've missed some strengths & weaknesses

**Comment by bnchdrff** [comment]
my opinion:

i think we should move forward with work on #42 -- we already have a lot of nice work done on it. even though developing a site using backbone can feel like getting lost in the woods, i personally like forests. a lot of the stuff we need to do is simple and might be *a little* harder when using backbone (i.e. template fragments that contain d3-rendered bits) but it's the same amount of difficulty as using any other frontend framework -- there isn't a framework that i've seen that is specifically tailored to a site like ours.

specifically:

#56 is already fixed -- that was an initial backbone app architecture oversight in my opinion - index.html isn't templated!

#49 and #54 are already fixed

#46 is irrelevant if we choose to finish the isomorphic work.. and in the meantime, memory is cheaper than my time

#53 is work whichever framework we use -- while some frameworks have sane defaults, you still need to tweak them or just rewrite your own

doing a full static render of the site is an interesting idea, until i start to think of the storage requirements as our # of grants and organizations increases -- i'd rather let a LRU cache manage this instead of needing to run a generator every time we push changes
